hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/MetricBaselineAggregationConverter.java

```diff
@@ -13,50 +14,22 @@
   }
 
   private MetricBaselineAggregation getBaselinedMetricAggregation(Baseline baseline) {
-    if (baseline == null) {
-      new MetricBaselineAggregationDefaultInstance();
+    if (Baseline.getDefaultInstance().equals(baseline)) {
+      return MetricBaselineAggregationImpl.EMPTY;
     }
-    return new MetricBaselineAggregationImpl(baseline);
+    return new MetricBaselineAggregationImpl(
+        baseline.getValue().getDouble(),
+        baseline.getLowerBound().getDouble(),
+        baseline.getUpperBound().getDouble());
   }
 
+  @lombok.Value
+  @Accessors(fluent = true)
   static class MetricBaselineAggregationImpl implements MetricBaselineAggregation {
-    private final Baseline baseline;
-
-    public MetricBaselineAggregationImpl(Baseline baseline) {
-      this.baseline = baseline;
-    }
-
-    @Override
-    public Double value() {
-      return baseline.getValue().getDouble();
-    }
-
-    @Override
-    public Double lowerBound() {
-      return baseline.getValue().getDouble();
-    }
-
-    @Override
-    public Double upperBound() {
-      return baseline.getValue().getDouble();
-    }
-  }
-
-  static class MetricBaselineAggregationDefaultInstance implements MetricBaselineAggregation {
-
-    @Override
-    public Double lowerBound() {
-      return null;
-    }
-
-    @Override
-    public Double upperBound() {
-      return null;
-    }
-
-    @Override
-    public Double value() {
-      return null;
-    }
+    static final MetricBaselineAggregation EMPTY =
```

Comment:
So this would end up with null values rather than a null baseline object - we can go with that and revisit